internal/storage/v2/elasticsearch/tracestore/from_dbmodel.go

```diff
@@ -25,22 +25,70 @@ import (
 var errType = errors.New("invalid type")
 
 // FromDBModel converts multiple ES DB Spans to internal traces
-func FromDBModel(spans []dbmodel.Span) (ptrace.Traces, error) {
+func FromDBModel(spans []dbmodel.Span, tagDotReplacement string) (ptrace.Traces, error) {
 	traceData := ptrace.NewTraces()
 	if len(spans) == 0 {
 		return traceData, nil
 	}
-
+	fromDb := &fromDBModel{tagDotReplacement: tagDotReplacement}
 	resourceSpans := traceData.ResourceSpans()
 	resourceSpans.EnsureCapacity(len(spans))
-	err := dbSpansToSpans(spans, resourceSpans)
+	err := fromDb.dbSpansToSpans(spans, resourceSpans)
 
 	return traceData, err
 }
 
-func dbProcessToResource(process dbmodel.Process, resource pcommon.Resource) {
+type fromDBModel struct {
+	tagDotReplacement string
+}
+
+func (f *fromDBModel) convertTagField(k string, v any) dbmodel.KeyValue {
+	dKey := strings.ReplaceAll(k, f.tagDotReplacement, ".")
+	dTag := dbmodel.KeyValue{Key: dKey, Value: v}
+	switch val := v.(type) {
+	case int64:
+		dTag.Type = dbmodel.Int64Type
+	case float64:
+		dTag.Type = dbmodel.Float64Type
+	case bool:
+		dTag.Type = dbmodel.BoolType
+	case string:
+		dTag.Type = dbmodel.StringType
+	case json.Number:
+		n, err := val.Int64()
+		if err == nil {
+			dTag.Type = dbmodel.Int64Type
+			dTag.Value = n
+		} else {
+			f, err := val.Float64()
+			if err == nil {
+				dTag.Type = dbmodel.Float64Type
+				dTag.Value = f
+			}
+		}
+	default:
+		dTag.Type = ""
+		dTag.Value = "invalid type"
+	}
+	return dTag
+}
+
+func (f *fromDBModel) mergeTagMapAndTagSlice(kvs []dbmodel.KeyValue, tagMap map[string]any) []dbmodel.KeyValue {
```

Comment:
utility function should not be at the top of the file, it breaks the flow of logic

Reply:
function name needs to reflect the business function, not restate its signature. What it's doing is merging nested and materialized tags.
```suggestion
func (f *fromDBModel) mergeNestedAndElevatedTags(kvs []dbmodel.KeyValue, tagMap map[string]any) []dbmodel.KeyValue {
```